doc/manual/core_device.rst

```diff
@@ -8,9 +8,77 @@ While it is possible to use the other parts of ARTIQ (controllers, master, GUI,
 .. _core-device-flash-storage:
 
 Flash storage
-^^^^^^^^^^^^^
-
-The core device contains some flash storage space which is used to store configuration data. It is one sector (typically 64 kB) large and organized as a list of key-value records, accessible by using ``artiq_coremgmt`` (see :ref:`core-device-management-tool`). The core device IP and MAC addresses, as well as, if present, the startup and/or idle kernels (see :ref:`miscellaneous_config_core_device`) are stored here. 
+-------------
+
+The core device contains some flash storage space which is used to store configuration data and the idle/startup kernels. It is one sector (typically 64 kB) large and organized as a list of key-value records, accessible either through ``artiq_mkfs`` and ``artiq_flash`` or (preferable in most cases) the ``config`` option of the ``artiq_coremgmt`` core management tool (see below). Information can be stored to keys of any name, but the specific keys currently used and referenced by ARTIQ in one or another form are summarized below: 
```

Comment:
"in one or another form" -> "in one form or another". Alternatively, remove it for conciseness.